android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -257,6 +258,16 @@ protected void onResume() {
     KMKeyboardDownloaderActivity.addKeyboardDownloadEventListener(this);
     PackageActivity.addKeyboardDownloadEventListener(this);
 
+    // Check if default dictionary model package should be installed
```

Comment:
Does this mean that the default lexical model will be reinstalled every time the app resumes, so the user can effectively never remove it?

Reply:
Ah, good point. I'll make this a one-time initialization preference that gets stored under the hood.

Reply:
Make sure you bundle that with the keyboard install check above, too :)

Reply:
I ended up making the default keyboard a separate preference (to ensure `addKeyboardDownloadEventListener` calls in between always happen)

And I had to check for valid current keyboard info before installing the default dictionary. Otherwise, the packageActivity wouldn't trigger.

Reply:
Screenshots showing the impact:
1. on initial startup, the model isn't loaded yet
![get started](https://user-images.githubusercontent.com/7358010/64148083-8b373c00-ce4c-11e9-9da2-4ff22558dec1.png)

2. After "Get Started" is dismissed, the app "resumes", installs the default dictionary, and loads the dictionary. There's some keyboard flicker during this transition
![loaded](https://user-images.githubusercontent.com/7358010/64148112-a1dd9300-ce4c-11e9-8a15-20e79bdda960.png)

Reply:
Alas, I can't find a way to install the default dictionary earlier.
It seems for the PackageActivity to run and trigger the event handlers, it needs to happen after "Get Started" is shown...